File: src/main/java/com/comcast/viper/hlsparserj/IPlaylist.java
package com.comcast.viper.hlsparserj;

import java.util.List;

import com.comcast.viper.hlsparserj.tags.UnparsedTag;
import com.comcast.viper.hlsparserj.tags.Version;

/**
 * Serves as the interface of a playlist object.  There are
 * two kinds of playlists that are supported:
 *
 * <ul>
 *  <li>master playlist: if URI lines in playlist identify media playlists</li>
 *  <li>media playlist: if URI lines in playlist identify media segments</li>
 * </ul>
 */
public interface IPlaylist {

    /**
     * Returns the version of the playlist.
     * @return version
     */
    Version getVersion();

    /**
     * Returns true if this is a master playlist; false if a media playlist.
     * @return boolean
     */
    boolean isMasterPlaylist();

    /**
     * Returns list of tags in this playlist.
     * @return list of tags
     */
    List<UnparsedTag> getTags();
}

File: src/main/kotlin/com/realeyes/hlsparserj/IPlaylist.kt
/**
*
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.realeyes.hlsparserj

import com.realeyes.hlsparserj.tags.UnparsedTag
import com.realeyes.hlsparserj.tags.Version

/**
 * Serves as the interface of a playlist object.  There are
 * two kinds of playlists that are supported:
 *
 *
 *  * master playlist: if URI lines in playlist identify media playlists
 *  * media playlist: if URI lines in playlist identify media segments
 *
 */
interface IPlaylist {
    /**
     * Returns the version of the playlist.
     * @return version
     */
    fun getVersion(): Version?

    /**
     * Returns true if this is a master playlist; false if a media playlist.
     * @return boolean
     */
    val isMasterPlaylist: Boolean

    /**
     * Returns list of tags in this playlist.
     * @return list of tags
     */
    val tags: List<UnparsedTag>
}
